### PR TITLE
Moving async-content-test from legacy tests to the basic app

### PR DESCRIPTION
**Pull request by kiwiupover**
Move async-content-test from legacy tests to the basic app tests
Delete legacy test

### CODE REVIEW

**Comment by kiwiupover** [comment]
@rwjblue can you review this please

File: test-packages/basic-app/app/components/async-content.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+//     let deferred = this.get('deferred');
+
+//     if (runLater) {
+//       Ember.run.cancel(runLater);
+//       deferred.reject('cancelled');
+//     }
+
+//     this._super(...arguments);
+//   }
+// });
```
Comment (rwjblue) [inline test-packages/basic-app/app/components/async-content.js:60]: Should be deleted or incorporated

File: test-packages/basic-app/test/async-content-test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  it('waits for async content when using `fastboot.deferRendering`', async () => {
+    const response = await request({
+      url: `http://localhost:45678/async-content`,
+      headers: {
+        'Accept': 'text/html'
+      }
+    });
+
+    console.log('body', response.body);
```
Comment (rwjblue) [inline test-packages/basic-app/test/async-content-test.js:27]: Probably don't need to keep this

**Review by rwjblue — CHANGES_REQUESTED** [review]

**Comment by kiwiupover** [comment]
@rwjblue I'm sorry I sent this over when it was not ready

It is ready now.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**